examples/nlp/gpt/conf/gpt_self_rewarding.yaml

```diff
@@ -0,0 +1,281 @@
+      useful way, regardless of whether it seems to have been written by an AI Assistant or if it
+      has elements typically found in blogs or search results.
+      - Grant a fourth point if the response is clearly written from an AI Assistant's perspective,
+      addressing the user's question directly and comprehensively, and is well-organized and helpful,
+      even if there is slight room for improvement in clarity, conciseness or focus.
+      - Bestow a fifth point for a response that is impeccably tailored to the user's question
+      by an AI Assistant, without extraneous information, reflecting expert knowledge, and
+      demonstrating a high-quality, engaging, and insightful answer.
+      
+      <prompt>{{ prompt }}<prompt>
+      <response>{{ response }}<response>
+      <judgement A>{{ judgement_a }}<judgement A>
+      <judgement B>{{ judgement_b }}<judgement B>
+      
+      After examining the original question, response, and both judgments:
+      - Explain which judgment is more accurate according to the original rubric and why.
+      Consider factors such as adherence to the rubric, accuracy in evaluating the response, and
+      consistency in applying the criteria.
+      - Conclude with a clear statement of which judgment is better using the format: "Winner: [Judgement A | Judgement B]"
+      <extra_id_1>Assistant
+
+
+    # params for generation
+    sampling_params:
+      use_greedy: False
+      temperature: 1.0
+      top_k: 0
+      top_p: 1.0
+      repetition_penalty: 1.0
+      add_BOS: False
+      all_probs: False
+      compute_logprob: False
+      end_strings: ["<|endoftext|>", "<extra_id_1>"]
+
+    # length argument for autoregressive sampling
+    # max length means max amount of tokens to generate
+    length_params:
+      max_length: ${int_div:${model.encoder_seq_length}, 2}
+      min_length: 1
+  
+  #encoder_seq_length: 4096
+  #max_position_embeddings: ${model.encoder_seq_length}
+
+  # miscellaneous
+  seed: 1234
+
+  optim:
+    name: distributed_fused_adam
+    bucket_cap_mb: 200
+    overlap_grad_sync: False
+    contiguous_grad_buffer: True
+    lr: 9e-6
+    weight_decay: 0.1 
+    betas: 
+    - 0.9
+    - 0.98
+    sched:
+      name: CosineAnnealing
+      warmup_steps: 10
+      constant_steps: 1000
+      min_lr: 9e-7
+
+  precision: ${trainer.precision}
+
+  data:
+    chat: True # whether use chatbot data or not
+    chat_prompt_tokens:  # special tokens for the chat prompts, a dictionary of {token_type: token}. note that some tokenizer may combine the characters at the junction between {end_of_turn}{turn_start}. e.g. '<im end><im start>', the '><' sometimes is merged to be a single token. This is not supported, try to avoid
+      system_turn_start: "<extra_id_0>"
+      turn_start: "<extra_id_1>"
+      label_start: "<extra_id_2>"
+      end_of_turn: "\x0A"  # \0x0A is '\n'
+      end_of_name: "\x0A"  # \0x0A is '\n'
```

Comment:
Since we have this, can we have a [meta-]judge prompt that doesn't include the `<extra_id>` stuff?

Reply:
Done and implemented

Reply:
Ok with me -- though I think we shouldn't even have the `System` / `User` / `Assistant` strings in the judge templates, but fixing this cleanly would probably mean addressing the pretty deep underlying issue of prompt templates in NeMo models... so let's not go there.